src/library_webgl.js

```diff
@@ -3086,6 +3091,72 @@ var LibraryGL = {
 
 #endif // ~GL_EXPLICIT_UNIFORM_LOCATION
 
+#if GL_EXPLICIT_UNIFORM_BINDING
+    // Extract the layout(binding = x) directives. Four types we need to handle:
+    // layout(binding = 3) uniform sampler2D mainTexture;
+    // layout(binding = 1, std140) uniform MainBlock { ... };
+    // layout(std140, binding = 1) uniform MainBlock { ... };
+    // layout(binding = 1) uniform MainBlock { ... };
+    var bindingRegex = /layout\s*\(.*?binding\s*=\s*([-\d]+).*?\)\s*uniform\s+(\w+)\s+(\w+)?/g, samplerBindings = {}, uniformBindings = {}, bindingMatch;
```

Comment:
Some word boundary markers would make this slightly more robust:

```suggestion
    var bindingRegex = /layout\s*\(.*?\bbinding\s*=\s*([-\d]+\b).*?\)\s*uniform\s+(\w+)\s+(\w+)?/g, samplerBindings = {}, uniformBindings = {}, bindingMatch;
```

Also why are hyphens allowed in the number? Can bindings be negative? If so, maybe:

```suggestion
    var bindingRegex = /layout\s*\(.*?\bbinding\s*=\s*(-?\d+\b).*?\)\s*uniform\s+(\w+)\s+(\w+)?/g, samplerBindings = {}, uniformBindings = {}, bindingMatch;
```

Reply:
Using `\b`s would certainly be more correct, but there is very very little chance that someone would have `layout(foolocation=x)` and be positively surprised that it works and it shouldn't. Reading regexes is mentally taxing, so every single added character increases cognitive load, so I think it will be better to omit `\b`s.

The note about `-?\d+` instead of `[-d\]` is a good one, that is certainly more correct.

> Also why are hyphens allowed in the number? Can bindings be negative?

This is to validate that people don't pass negative numbers. It is kind of silly, and in practice I doubt anyone will actually have this, so could certainly leave that out as well on the basis of the argument above. But thought to put that there since the native GL side extension spec explicitly words that out and forbids negative locations.